features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/GraphQLConsole/GraphQLAuthentication.jsx

```diff
@@ -0,0 +1,392 @@
+        margin: theme.spacing(1),
+        padding: theme.spacing(1),
+        height: theme.spacing(100),
+    },
+    titleSub: {
+        marginLeft: theme.spacing(2),
+        paddingTop: theme.spacing(2),
+        paddingBottom: theme.spacing(2),
+    },
+    root: {
+        margin: theme.spacing(1),
+        padding: theme.spacing(1),
+    },
+    grid: {
+        marginTop: theme.spacing(4),
+        marginBottom: theme.spacing(4),
+        paddingRight: theme.spacing(2),
+        justifyContent: 'center',
+    },
+    userNotificationPaper: {
+        padding: theme.spacing(2),
+    },
+    centerItems: {
+        margin: 'auto',
+    },
+    inputAdornmentStart: {
+        minWidth: theme.spacing(18),
+    },
+}));
+
+
+/**
+ *
+ * @param {*} props
+ */
+export default function GraphQLAuthentication(props) {
+    const classes = useStyles();
+
+    const {
+        api,
+        accessToken,
+        setAccessTocken,
+        authorizationHeader,
+        securitySchemeType,
+        setSecuritySchemeType,
+        prefix,
+        isApiKeyEnabled,
+        selectedEnvironment,
+        setSelectedEnvironment,
+        environments,
+        setURLss,
+        environmentObject,
+        setFound,
+    } = props;
+
+    const user = AuthManager.getUser();
+    const [showToken, setShowToken] = useState(false);
+    const [subscriptions, setSubscriptions] = useState(null);
+    const [selectedApplication, setSelectedApplication] = useState();
+    const [selectedKeyType, setSelectedKeyType] = useState('PRODUCTION');
+    const [keys, SetKeys] = useState();
+    const isPrototypedAPI = api.lifeCycleStatus && api.lifeCycleStatus.toLowerCase() === 'prototyped';
+
+    useEffect(() => {
+        const apiID = api.id;
+        const apiClient = new Api();
+        const promiseGraphQL = apiClient.getGraphQLSchemaByAPIId(apiID);
+
+        promiseGraphQL
+            .then(() => {
+                if (user != null) {
+                    return apiClient.getSubscriptions(apiID);
+                } else {
+                    return null;
+                }
+            })
+            .then((subscriptionsResponse) => {
+                if (subscriptionsResponse != null) { //
+                    const subs = subscriptionsResponse.obj.list.filter((item) => item.status === 'UNBLOCKED' || item.status === 'PROD_ONLY_BLOCKED');
+                    if (subs && subs.length > 0) {
+                        const sApplication = subs[0].applicationId;
+                        setSelectedApplication(sApplication);
+                        const promiseApp = Application.get(sApplication);
+                        promiseApp
+                            .then((application) => {
+                                return application.getKeys();
+                            })
+                            .then((appKeys) => {
+                                if (appKeys.get('SANDBOX')) {
+                                    const sKType1 = 'SANDBOX';
```

Comment:
I think skType1 and skType2 are redundant declarations. We can remove them or, define as constants.